test/unit/test_fiat.py

```diff
@@ -149,18 +167,72 @@ def __init__(self, a, b):
     "BrezziDouglasMarini(S, 1)",
     "BrezziDouglasMarini(S, 2)",
     "BrezziDouglasMarini(S, 3)",
+    'BrezziDouglasMarini(T, 1, variant="integral")',
+    'BrezziDouglasMarini(T, 2, variant="integral")',
+    'BrezziDouglasMarini(T, 3, variant="integral")',
+    'BrezziDouglasMarini(S, 1, variant="integral")',
+    'BrezziDouglasMarini(S, 2, variant="integral")',
+    'BrezziDouglasMarini(S, 3, variant="integral")',
+    'BrezziDouglasMarini(T, 1, variant="integral(2)")',
+    'BrezziDouglasMarini(T, 2, variant="integral(3)")',
+    'BrezziDouglasMarini(T, 3, variant="integral(4)")',
+    'BrezziDouglasMarini(S, 1, variant="integral(2)")',
+    'BrezziDouglasMarini(S, 2, variant="integral(3)")',
+    'BrezziDouglasMarini(S, 3, variant="integral(4)")',
+    'BrezziDouglasMarini(T, 1, variant="point")',
+    'BrezziDouglasMarini(T, 2, variant="point")',
+    'BrezziDouglasMarini(T, 3, variant="point")',
+    'BrezziDouglasMarini(S, 1, variant="point")',
+    'BrezziDouglasMarini(S, 2, variant="point")',
+    'BrezziDouglasMarini(S, 3, variant="point")',
     "Nedelec(T, 1)",
     "Nedelec(T, 2)",
     "Nedelec(T, 3)",
     "Nedelec(S, 1)",
     "Nedelec(S, 2)",
     "Nedelec(S, 3)",
+    'Nedelec(T, 1, variant="integral")',
+    'Nedelec(T, 2, variant="integral")',
+    'Nedelec(T, 3, variant="integral")',
+    'Nedelec(S, 1, variant="integral")',
+    'Nedelec(S, 2, variant="integral")',
+    'Nedelec(S, 3, variant="integral")',
+    'Nedelec(T, 1, variant="integral(2)")',
+    'Nedelec(T, 2, variant="integral(3)")',
+    'Nedelec(T, 3, variant="integral(4)")',
+    'Nedelec(S, 1, variant="integral(2)")',
+    'Nedelec(S, 2, variant="integral(3)")',
+    'Nedelec(S, 3, variant="integral(4)")',
+    'Nedelec(T, 1, variant="point")',
+    'Nedelec(T, 2, variant="point")',
+    'Nedelec(T, 3, variant="point")',
+    'Nedelec(S, 1, variant="point")',
+    'Nedelec(S, 2, variant="point")',
+    'Nedelec(S, 3, variant="point")',
     "NedelecSecondKind(T, 1)",
     "NedelecSecondKind(T, 2)",
     "NedelecSecondKind(T, 3)",
     "NedelecSecondKind(S, 1)",
     "NedelecSecondKind(S, 2)",
     "NedelecSecondKind(S, 3)",
+    'NedelecSecondKind(T, 1, variant="integral")',
+    'NedelecSecondKind(T, 2, variant="integral")',
+    'NedelecSecondKind(T, 3, variant="integral")',
+    'NedelecSecondKind(S, 1, variant="integral")',
+    'NedelecSecondKind(S, 2, variant="integral")',
+    'NedelecSecondKind(S, 3, variant="integral")',
+    'NedelecSecondKind(T, 1, variant="integral(2)")',
+    'NedelecSecondKind(T, 2, variant="integral(3)")',
+    'NedelecSecondKind(T, 3, variant="integral(4)")',
+    'NedelecSecondKind(S, 1, variant="integral(2)")',
+    'NedelecSecondKind(S, 2, variant="integral(3)")',
+    'NedelecSecondKind(S, 3, variant="integral(4)")',
```

Comment:
I find it kind of ugly that this has to parse expression encoded in strings.  Is there anything in the pipeline that asserts that `variant` must be a string?  Perhaps it would be better to loosen that assumption, and simply use proper symbolic expressions...

Reply:
UFL does this: https://github.com/FEniCS/ufl/blob/master/ufl/finiteelement/finiteelement.py#L169

Asserting that the variant hint is empty or a string. I agree that providing more structured information would be good, but I think that necessitates a somewhat wider refactoring of the way finite element objects are constructed/represented in UFL.

Reply:
Preferably, it should be something castable to a string, rather a string itself.